File: app/src/main/java/com/sudoajay/duplication_data/BackgroundProcess/WorkMangerProcess1.java
package com.sudoajay.duplication_data.BackgroundProcess;


import android.content.Context;
import android.os.Environment;
import androidx.annotation.NonNull;
import android.view.View;

import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.sudoajay.duplication_data.DuplicationData.ScanDuplicateData;
import com.sudoajay.duplication_data.DuplicationData.ShowDuplicate;
import com.sudoajay.duplication_data.Notification.NotifyNotification;
import com.sudoajay.duplication_data.Permission.AndroidSdCardPermission;
import com.sudoajay.duplication_data.R;
import com.sudoajay.duplication_data.sharedPreferences.TraceBackgroundService;

import java.io.File;
import java.util.ArrayList;


public class WorkMangerProcess1 extends Worker {


    public WorkMangerProcess1(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);

    }

    @NonNull
    @Override
    public Result doWork() {
            GetWorkDone(getApplicationContext());
        return Result.success();
    }

    public static void GetWorkDone(final Context context){
        // local variable
        int sdcard, internal = View.INVISIBLE;
        long size = 0;
        String textPass;

        ArrayList<String> savePath = new ArrayList<>();
        AndroidSdCardPermission androidSdCardPermission
                = new AndroidSdCardPermission(context);
        if (new File(Environment.getExternalStorageDirectory().getAbsolutePath()).exists()) {
            savePath.add(Environment.getExternalStorageDirectory().getAbsolutePath());
            internal = View.VISIBLE;
        }
        if (androidSdCardPermission.isSdStorageWritable()) {
            savePath.add(androidSdCardPermission.getSd_Card_Path_URL());
            sdcard = View.VISIBLE;
        } else {
            savePath.add(null);
            sdcard = View.INVISIBLE;
        }

        ScanDuplicateData scanDuplicateData = new ScanDuplicateData(context);
        scanDuplicateData.Duplication(savePath.get(0), savePath.get(1), internal, sdcard);

        for (String path : scanDuplicateData.getList()) {
            if (!path.equalsIgnoreCase("and")) {
                size += new File(path).length();
            }
        }

        if (size != 0) {
            textPass = "We Have Found " + ShowDuplicate.Convert_It(size) + " Of Duplicate Files";
            NotifyNotification notifyNotification = new NotifyNotification(context);
            notifyNotification.notify(textPass, context.getString(R.string.file_found_title));
        }


        TraceBackgroundService traceBackgroundService = new TraceBackgroundService(context);
        // set next date
        traceBackgroundService.setTaskA();
    }

}

File: app/src/main/java/com/sudoajay/duplication_data/backgroundProcess/WorkMangerProcess1.kt
package com.sudoajay.duplication_data.backgroundProcess

import android.content.Context
import android.os.Build
import android.view.View
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.sudoajay.duplication_data.R
import com.sudoajay.duplication_data.duplicationData.ExpandableDuplicateListAdapter
import com.sudoajay.duplication_data.duplicationData.ScanDuplicateDataWithDoc
import com.sudoajay.duplication_data.duplicationData.ScanDuplicateDataWithFile
import com.sudoajay.duplication_data.helperClass.FileSize
import com.sudoajay.duplication_data.notification.NotifyNotification
import com.sudoajay.duplication_data.permission.AndroidExternalStoragePermission
import com.sudoajay.duplication_data.permission.AndroidSdCardPermission
import com.sudoajay.duplication_data.sharedPreferences.TraceBackgroundService
import java.io.File
import java.util.*

class WorkMangerProcess1(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
    override fun doWork(): Result {
        getWorkDone(applicationContext)
        return Result.success()
    }

    companion object {
        fun getWorkDone(context: Context) { // local variable
            var sdcardCheck = View.INVISIBLE
            var internalCheck = View.INVISIBLE
            var fileSize: Long = 0
            val textPass: String
            val listHeaderChild: LinkedHashMap<Int, List<String>>
            val androidSdCardPermission = AndroidSdCardPermission(context)


            if (File(AndroidExternalStoragePermission.getExternalPath(context).toString()).exists())
                internalCheck = View.VISIBLE

            if (androidSdCardPermission.isSdStorageWritable) sdcardCheck = View.VISIBLE


            //  Here use of DocumentFile in android 10 not File is using anymore
            listHeaderChild = if (Build.VERSION.SDK_INT <= 29) {
                val scanDuplicateDataWithFile = ScanDuplicateDataWithFile(context)
                scanDuplicateDataWithFile.duplication(internalCheck, sdcardCheck)
                scanDuplicateDataWithFile.listHeaderChild

            }else{
                val scanDuplicateDataWithDoc = ScanDuplicateDataWithDoc(context)
                scanDuplicateDataWithDoc.duplication(internalCheck, sdcardCheck)
                scanDuplicateDataWithDoc.listHeaderChild

            }

            for (lists in listHeaderChild.values) {
                for (i in lists){
                    fileSize += ExpandableDuplicateListAdapter.getFileSizeInBytes(i, context)
                }
            }

            if (fileSize != 0L) {
                textPass = "We Have Found " + FileSize.convertIt(fileSize) + " Of unnecessary Files"
                val notifyNotification = NotifyNotification(context)
                notifyNotification.notify(textPass, context.getString(R.string.file_found_title))
            }
            val traceBackgroundService = TraceBackgroundService(context)
            // set next date
            traceBackgroundService.setTaskA()
        }
    }
}